xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPIGL.cpp

```diff
@@ -40,13 +40,13 @@ CBaseRenderer* CRendererVAAPI::Create(CVideoBuffer *buffer)
   return nullptr;
 }
 
-void CRendererVAAPI::Register(IVaapiWinSystem *winSystem, VADisplay vaDpy, EGLDisplay eglDisplay, bool &general, bool &deepColor)
+bool CRendererVAAPI::Register(IVaapiWinSystem *winSystem, VADisplay vaDpy, EGLDisplay eglDisplay, bool &general, bool &deepColor)
 {
   int major_version, minor_version;
   if (vaInitialize(vaDpy, &major_version, &minor_version) != VA_STATUS_SUCCESS)
   {
     vaTerminate(vaDpy);
-    return;
+    return false;
   }
 
   general = deepColor = false;
```

Comment:
Wouldn't moving this a few lines up also solve this?

Reply:
How?

Reply:
Your stack trace tells that `general` isn't initialised. So move it up to also set them to false if `vaInitialize` fails.

Reply:
So by "this" you meant the assignment of the two variables? Yes, one could do that, but since those parameters were not documented, I could not figure out their meaning, and was not sure whether this assignment after a `vaInitialize()` failure was a valid state. Is it?

Reply:
By "this" I meant the line I commented on. Sorry if it wasn't obvious.
The variables describe whether vaapi is available (`general`) and whether deep color rendering is available (`deepColor`). So yes, having both set to `false` is the expected state when vaapi initialization fails.

Reply:
Oh, then this code was more inconsistent and confusing than I thought. Because the line you commented on wasn't even necessary at all previously - `CVaapi?Texture::TestInterop()` already does that. That's what you get for putting return values in reference parameters...
The real fix (without fixing the API design) would be to move the assignments into the `vaInitialize()` error branch.

Reply:
> The real fix (without fixing the API design) would be to move the assignments into the vaInitialize() error branch.

I think moving the assignment to the top of the function is clearer and doesn't hurt.

@FernetMenta I also very much dislike the non-const reference parameters. Can we change this?

Reply:
My idea for a proper API design: throw exception on error, and on success, return additional information (e.g. `deepColor`). But never return values in reference parameters.